akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2Multiplexer.scala

```diff
@@ -286,21 +286,21 @@ private[http2] trait Http2MultiplexerSupport { logic: GraphStageLogic with Stage
     var timings = Map.empty[String, Long].withDefaultValue(0L)
     var lastTimestamp = System.nanoTime()
 
-    def recordStateChange(oldState: String, newState: String): Unit = {
+    def recordStateChange(oldState: String, newState: String): Unit = debug {
       val now = System.nanoTime()
       val lasted = now - lastTimestamp
       val name = oldState
       timings = timings.updated(name, timings(name) + lasted)
       lastTimestamp = now
-      debug(s"Changing state from $oldState to $newState")
+      s"Changing state from $oldState to $newState"
     }
 
     /** Logs DEBUG level timing data for the output side of the multiplexer*/
-    def reportTimings(): Unit = {
+    def reportTimings(): Unit = debug {
       val timingsReport = timings.toSeq.sortBy(_._1).map {
         case (name, nanos) => f"${nanos / 1000000}%5d ms $name"
       }.mkString("\n")
-      debug(s"Timing data for connection\n$timingsReport")
+      s"Timing data for connection\n$timingsReport"
```

Comment:
it's not nice to have line breaks in logs, but that is another matter

Reply:
That part is currently not even used anywhere :)